dmscripts/notify_suppliers_of_awarded_briefs.py

```diff
@@ -0,0 +1,86 @@
+from datetime import datetime, timedelta
+
+from dmutils.email.exceptions import EmailError
+from dmutils.email.helpers import hash_string
+from dmutils.formats import DATE_FORMAT
+from dmscripts.helpers import env_helpers
+
+
+def _create_context_for_brief(stage, brief):
+    return {
+        'brief_title': brief['title'],
+        'brief_link': '{}/{}/opportunities/{}'.format(
+            env_helpers.get_web_url_from_stage(stage),
+            brief['framework']['family'],
+            brief['id']
+        )
+    }
+
+
+def main(
+    data_api_client, mail_client, template_id, stage, logger,
+    dry_run=False, brief_response_ids=None, awarded_at=None
+):
+    failed_brief_responses, brief_responses = [], []
+
+    if brief_response_ids:
+        # Used for re-running failed email sending
+        for brief_response_id in brief_response_ids:
+            brief_responses.append(data_api_client.get_brief_response(brief_response_id)['briefResponses'])
+    else:
+        awarded_at_date = awarded_at or (datetime.utcnow() - timedelta(days=1)).strftime(DATE_FORMAT)
+        awarded_brief_responses = data_api_client.find_brief_responses_iter(awarded_at=awarded_at_date)
```

Comment:
So... if the script fails to run one night, those alerts are just never sent out, are they? Similarly, the script gives different results if it's run one minute before midnight or one minute after midnight? I guess I don't have a better answer without recording state back on the server somehow...

Reply:
We'll get a failure alert in Slack, and we can re-run the job for a specific date. 

Our other emailing scripts have the convention of running between 8am-9am about stuff that happened the day before (so the buyer has until 23:59:59.999999 UTC to award a brief in order for an email to get sent the next day).